Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                newtsStep.StepID = testcaseStep.Id.ToString();
+                newtsStep.StepName = testcaseStep.Name;
+                newtsStep.Description = testcaseStep.Description;
+                newtsStep.Expected = "";// testcaseStep.ElementsField["expected"].ToString();
+                newTSTest.Steps.Add(newtsStep);
+                //Get the TC parameters
+                CheckForParameter(newTSTest, newtsStep.Description);
+            }
+
+            //Get the TC execution history
+            try
+            {
+                List<RunSuite> TSTestRuns = GetTestSuiteRun(testCase.TestSetId);
+
+                foreach (RunSuite run in TSTestRuns)
+                {
+                    QC.QCTSTestRun newtsRun = new QC.QCTSTestRun();
+                    newtsRun.RunID = run.Id;
+                    newtsRun.RunName = run.Name;
+                    newtsRun.Status = run.NativeStatus.Name;
+                    newtsRun.ExecutionDate = (run.GetValue("started").ToString());
+                    newtsRun.ExecutionTime = (run.GetValue("last_modified").ToString());
+                    newtsRun.Tester = (run.DefaultRunBy.FullName).ToString();
+                    newTSTest.Runs.Add(newtsRun);
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to pull QC test case RUN info", ex);
+                newTSTest.Runs = new List<QC.QCTSTestRun>();
+            }
+            return newTSTest;
+        }
+
+        public List<RunSuite> GetTestSuiteRun(string testSuiteId)
+        {
+            QCRunColl runColl = new QCRunColl();
+            EntityListResult<RunSuite> testFolders = new EntityListResult<RunSuite>();
+            LogicalQueryPhrase test_Suite = new LogicalQueryPhrase("id", testSuiteId, ComparisonOperator.Equal);
+            CrossQueryPhrase qd = new CrossQueryPhrase("test", test_Suite);
+
+            return octaneRepository.GetEntities<RunSuite>(GetLoginDTO(), new List<IQueryPhrase> { qd });
+        }
+
+        public dynamic GetTSRunStatus(dynamic TSItem)
+        {
+            List<RunSuite> testInstances = GetTestSuiteRun(TSItem.TestSetID);
+
+            foreach (RunSuite testInstance in testInstances)
+            {
+                bool existing = false;
+                foreach (string[] status in TSItem.TestSetStatuses)
+                {
+                    if (status[0] == testInstance.NativeStatus.Name)
+                    {
+                        existing = true;
+                        status[1] = (Int32.Parse(status[1]) + 1).ToString();
+                    }
+                }
+                if (!existing) { TSItem.TestSetStatuses.Add(new string[] { testInstance.NativeStatus.Name, "1" }); }
+            }
+            return TSItem;
+        }
+
+        public BusinessFlow ConvertQCTestSetToBF(QC.QCTestSet testSet)
+        {
+            try
+            {
+                if (testSet == null) return null;
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)